src/components/Tag/Tag.jsx

```diff
@@ -60,8 +60,15 @@ const Tag = createClass({
 		const { isRemovable, children, className, ...passThroughs } = this.props;
 
 		const subTags = filterTypes(children, Tag);
+
 		const otherChildren = rejectTypes(children, Tag);
+
 		const hasOtherChildren = !_.isEmpty(otherChildren);
+
+		const hasGrandChildren = !!subTags.find(
+			tag => parseInt(tag.key.slice(1, 2)) > 1
```

Comment:
I'm a little confused by this, what is `tag.key` and how is it determining that there are tag grand children? I'd imagine we'd need something like `filterTypes(subTags, Tag).length > 1` or something (not tested).

Reply:
The tags are from [React.children.toArray()](https://reactjs.org/docs/react-api.html#reactchildrentoarray) and the keys have a weird string representation, but if the second character is greater than 1, then it has a grandchild.

That's true, we could loop through the subTags too. That would be more readable too.